File: det.py
def determination(matrix):
  if (len(matrix)!=2 or len(matrix[1])!=2):
     raise valueerror("input matrix must be a 2*2 matrix")
  return matrix[0][0]*matrix[1][1]-matrix[0][1]
matrix_2=[[3,4],[3,4]]
result=determination(matrix_2)
print("Determination of 2*2 matrix",result)

File: det.m
function result = determination(matrix)
    if (size(matrix) != [2, 2])
        error("Input matrix must be a 2x2 matrix");
    end
    result = matrix(1,1) * matrix(2,2) - matrix(1,2) * matrix(2,1);
end

matrix_2 = [3, 4; 3, 4];
result = determination(matrix_2);
disp(["Determination of 2x2 matrix: ", num2str(result)]);